crates/uv-interpreter/src/discovery.rs

```diff
@@ -0,0 +1,1134 @@
+        return false;
+    }
+
+    let mut buf = [0u16; MAXIMUM_REPARSE_DATA_BUFFER_SIZE as usize];
+    let mut bytes_returned = 0;
+
+    // SAFETY: The buffer is large enough to hold the reparse point.
+    #[allow(unsafe_code, clippy::cast_possible_truncation)]
+    let success = unsafe {
+        DeviceIoControl(
+            reparse_handle,
+            FSCTL_GET_REPARSE_POINT,
+            std::ptr::null_mut(),
+            0,
+            buf.as_mut_ptr().cast(),
+            buf.len() as u32 * 2,
+            &mut bytes_returned,
+            std::ptr::null_mut(),
+        ) != 0
+    };
+
+    // SAFETY: The handle is valid.
+    #[allow(unsafe_code)]
+    unsafe {
+        CloseHandle(reparse_handle);
+    }
+
+    // If the operation failed, assume it's not a reparse point.
+    if !success {
+        return false;
+    }
+
+    let reparse_point = String::from_utf16_lossy(&buf[..bytes_returned as usize]);
+    reparse_point.contains("\\AppInstallerPythonRedirector.exe")
+}
+
+/// On Unix, we do not need to deal with Windows store shims.
+///
+/// See the Windows implementation for details.
+#[cfg(not(windows))]
+fn is_windows_store_shim(_path: &Path) -> bool {
+    false
+}
+
+impl InterpreterRequest {
+    /// Create a request from a string.
+    ///
+    /// This cannot fail, which means weird inputs will be parsed as [`InterpreterRequest::File`] or [`InterpreterRequest::ExecutableName`].
+    pub fn parse(value: &str) -> Self {
+        // e.g. `3.12.1`
+        if let Ok(version) = VersionRequest::from_str(value) {
+            return Self::Version(version);
+        }
+        // e.g. `python3.12.1`
+        if let Some(remainder) = value.strip_prefix("python") {
+            if let Ok(version) = VersionRequest::from_str(remainder) {
+                return Self::Version(version);
+            }
+        }
+        // e.g. `pypy@3.12`
+        if let Some((first, second)) = value.split_once('@') {
+            if let Ok(implementation) = ImplementationName::from_str(first) {
+                if let Ok(version) = VersionRequest::from_str(second) {
+                    return Self::ImplementationVersion(implementation, version);
+                }
+            }
+        }
+        for implementation in ImplementationName::iter() {
+            if let Some(remainder) = value
+                .to_ascii_lowercase()
+                .strip_prefix(implementation.as_str())
+            {
+                // e.g. `pypy`
+                if remainder.is_empty() {
+                    return Self::Implementation(*implementation);
+                }
+                // e.g. `pypy3.12`
+                if let Ok(version) = VersionRequest::from_str(remainder) {
+                    return Self::ImplementationVersion(*implementation, version);
+                }
+            }
+        }
+        let value_as_path = PathBuf::from(value);
+        // e.g. ./path/to/.venv
+        if value_as_path.is_dir() {
```

Comment:
Honestly I think the path checks should be first. If you have `python3` in the current working directory, I'd expect `./python3` and `python3` to work the same (as they would if you ran them directly from the shell).

Reply:
I don't think that's how shells usually work though:

```
❯ touch foo
❯ chmod +x foo
❯ foo
zsh: command not found: foo
❯ ./foo
❯ bash
bash-5.2$ foo
bash: foo: command not found
bash-5.2$ ./foo
```

Reply:
I changed this behavior to match Charlie's expectation.